### PR TITLE
fix(scheduler): Compute next runs from structured intent

### PR DESCRIPTION
**Pull request by dcramer**
Scheduler create and update tools now accept structured one-off or recurring
intent instead of model-computed timestamps. The scheduler materializes
`next_run_at` from its trusted clock and timezone rules, including explicit DST
gap/fold behavior and recurrence anchoring. Existing one-off and recurring user
features remain available, while the legacy top-level `next_run_at`,
`schedule_kind`, and `recurrence` inputs are intentionally removed without a
compatibility shim.

Create tool-call identities derive stable task ids and use atomic
create-if-absent storage, so concurrent retries of the same call return one
committed task. Separate create calls remain independent and may intentionally
produce equivalent tasks. The persisted task shape is unchanged and no SQL
migration is needed.

The main review surface is the structured intent compiler and cadence logic;
the Slack integration tests and scheduler evals cover the model-facing hard cut,
rescheduling, create retry recovery, DST transitions, and calendar edge cases,
including multi-week schedules.

### CODE REVIEW

File: packages/junior-scheduler/src/schedule-intent.ts [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+    timezone,
+  });
+  if (nextRunAtMs === undefined) {
+    throw new ScheduleIntentError(
+      "The recurring schedule has no valid future occurrence.",
+    );
+  }
+  const materializedRecurrence = args.intent.start_date
+    ? recurrence
+    : { ...recurrence, startDate: localDateAt(nextRunAtMs, timezone) };
```
[the remark this answers is not in the document]
Comment (dcramer) [reply packages/junior-scheduler/src/schedule-intent.ts:362]: This behavior is intentional: when start_date is omitted, the next matching occurrence becomes the recurrence anchor, and the requested interval applies from that first run. There is no earlier N-day boundary to preserve. I added a regression asserting that an unanchored daily interval of 2 starts at the next occurrence and then advances by two days. While checking the adjacent interval behavior, I also fixed multi-week schedules to use Monday-based calendar-week boundaries.

*— Claude Code*

File: packages/junior-scheduler/src/tool-support.ts [hunk 2]
```diff
@@ -31,6 +29,37 @@ const TASK_ID_PREFIX = "sched";
...
+    );
+  }
+  return value;
+}
+
+function fingerprint(value: unknown): string {
+  return createHash("sha256")
+    .update(JSON.stringify(canonicalFingerprintValue(value)))
+    .digest("hex");
+}
```
[the remark this answers is not in the document]
Comment (dcramer) [reply packages/junior-scheduler/src/tool-support.ts]: Agreed that comparing semantically equivalent argument spellings is unnecessary complexity here. I removed argument fingerprints and the update mutation ledger entirely. A scoped toolCallId is now the sole create retry identity: replaying that call returns the committed task, while a separate tool call may intentionally create an equivalent task. Persisted task records no longer gain retry metadata.

*— Claude Code*

File: packages/junior-scheduler/src/tools/create-task.ts [hunk 3]
```diff
@@ -27,99 +28,95 @@ export function createSlackScheduleCreateTaskTool(
...
+        if (
+          !sameDestination(existing, destination) ||
+          existing.createdBy.slackUserId !== actor.slackUserId
+        ) {
+          throwToolInputError("Scheduled task operation identity is invalid.");
+        }
+        return scheduleTaskToolResult(
+          "slackScheduleCreateTask",
+          compactTask(existing),
+        );
```
[the remark this answers is not in the document]
Comment (dcramer) [reply packages/junior-scheduler/src/tools/create-task.ts:70]: This is intentional tool-call idempotency. Replaying the same historical create operation should return its current task, including a later deletion; it must not undo that deletion. A user asking to create the task again produces a new toolCallId, and separate create calls are explicitly allowed to create equivalent active tasks (covered by the adjacent integration test). Resurrecting on replay would make a transport retry reverse a subsequent user action.
